enhancements/authentication/pod-security-admission-autolabeling.md

```diff
@@ -0,0 +1,290 @@
+a mechanism that would synchronize the PSa policy level for a namespace that workload runs
+in so that the workload is allowed by both SCC and PSa admissions. For that purpose, a PSa
+Label Synchronization Controller (hereinafter the Controller) shall be built.
+
+The Controller watches changes of:
+- **namespaces**
+  - manual changes to the interesting labels of the namespaces that were
+    opted-in to label synchronization should be reverted
+- all known **workload types** in controlled namespaces
+  - new pods may be getting rejected by PSa for some time before the label
+    synchronization happens
+  - the controller should not allow more relaxed PSa admission level than
+    what SCCs allow
+- **roles** and **rolebindings** (and their **cluster variants**)
+  - to be able to properly assess SA capabilities in controlled namespaces
+- **SCCs**
+  - to properly assess restrictive PSa levels per affected namespaces
+
+The generic control loop of the Controller performs the following actions:
+1. list all controlled (opted-in by a label) namespaces
+2. for each namespace:
+    1. get all instances of known workload types
+    1. associate SCCs to the workloads based on the SAs running them, create a
+       SCC <-> []SA associations
+    1. order observed SCCs most-privileged first (according to PSa privilege levels
+       they map to)
+    1. evaluate the namespace's PSa privilege level based on the most privileged
+       SCC that is allowed to be used by its SA
+    1. modify the namespace's PSa labels based on the enforce level obsedved in the
+       previous step
+
+There is another control loop reacting only to changes of namespaces which performs
+all the steps in 2.
+
+#### Label Synchronization Opting-in
+
+The Controller described in [PSa Label Synchronization Controller](#psa-label-synchronization-controller)
+works on an opt-in basis, meaning that users need to specifically sign themselves up
+if they wanted to stick with just SCCs.
+
+The opt-in is driven by the `security.openshift.io/scc.podSecurityLabelSync: “true”`
+label, namespaces containing it will have their PSa labels reconciled by
+the Controller.
+
+The opt-in nature of the Controller would cause current workloads to break.
+The choice of opting-in is therefore only left upon the users that are
+capable of creating the `Namespace` objects directly. Such a user can clearly
+set the appropriate labels themselves.
+
+The rest of the users rely on `Namespace` creation delegated via the OpenShift
+`Project` API. The default `Project` template should therefore be modified so
+that it by default includes the `security.openshift.io/scc.podSecurityLabelSync: “true”`
+label. This ensures that the decision to use PSa as the main authorization
+mechanism is left to privileged users only and the workloads of the rest
+of the users keeps working.
+
+Simply updating the `Project` template is not enough to keep older workloads
```

Comment:
This is critical information that you should start with.

Reply:
I'd suggest framing the proposal such that it's clear that this will be done in 2 stages, where:
1. stage 1 (n-1) - introduce this functionality mode in non-breaking mode
2. stage 2 (n) - introduce this in breaking mode
is that accurate or am I misreading something?

Reply:
Additionally, I'm missing answers to the following:
1. I'm a current SCC user how that impacts me?
2. I haven't used SCC in the past and I don't care, does this impacts me?
3. I want to use PS over SCC how do I do that? 
Not sure if I missed something else, but it would be good to describe different options.